### PR TITLE
omap updates, kernel 4.19, uboot, emmc boot

### PR DESCRIPTION
**Pull request by dhewg**
This updates uboot and the kernel, which is the first combination to boot reliably off emmc on my boneblack.

Booting off emmc is archived by switching to uboot's generic distro boot. Supporting other devices, like nand, should be simple to add on top of this.

Tested on boneblack.

~~uboot is current at v2019-10-rc4. I'd update this MR once the final release is out.~~
Maybe someone could test this on `beagle` and `panda`, I lack the hardware to do so.

### CODE REVIEW

**Comment by dhewg** [comment]
Update:
* update uboot to v2019-10 release
* add patch to remove am33x-cm3

File: target/linux/omap/config-4.19 [hunk 1]
```diff
@@ -184,9 +197,11 @@ CONFIG_DEBUG_INFO=y
 CONFIG_DEBUG_LL_INCLUDE="mach/debug-macro.S"
 CONFIG_DEBUG_LOCK_ALLOC=y
 CONFIG_DEBUG_MUTEXES=y
+CONFIG_DEBUG_RT_MUTEXES=y
+CONFIG_DEBUG_RWSEMS=y
 CONFIG_DEBUG_SPINLOCK=y
```
Comment (hauke) [inline target/linux/omap/config-4.19]: Do we need all these debug support?

File: target/linux/omap/config-4.19 [hunk 2]
```diff
@@ -666,25 +699,32 @@ CONFIG_THERMAL_GOV_USER_SPACE=y
...
+# CONFIG_TI_EMIF_SRAM is not set
 CONFIG_TI_PIPE3=y
 CONFIG_TI_SOC_THERMAL=y
+CONFIG_TI_SYSC=y
 CONFIG_TI_THERMAL=y
 CONFIG_TMPFS_POSIX_ACL=y
+CONFIG_TRACEPOINTS=y
 CONFIG_TRACE_CLOCK=y
 CONFIG_TRACE_IRQFLAGS=y
+CONFIG_TRACING=y
```
Comment (hauke) [inline target/linux/omap/config-4.19]: Do we need this?

**Review by hauke — APPROVED** [review]

File: target/linux/omap/config-4.19 (hunk 1 above)
Comment (dhewg) [reply target/linux/omap/config-4.19]: I don't think so, most of those are set because of CONFIG_PROVE_LOCKING from 68ff0d3d788eeb1880c997a8a9053bb6fdc37755. I'll add a patch to get rid of it

Comment (dhewg) [reply target/linux/omap/config-4.19]: Done, removed

File: target/linux/omap/config-4.19 (hunk 2 above)
Comment (dhewg) [reply target/linux/omap/config-4.19]: Same, removed

**Comment by dhewg** [comment]
Anything I can do to get this in?

**Comment by lynxis** [comment]
@dhewg sorry for the delay. i'll do it tomorrow.

**Comment by lynxis** [comment]
hi @dhewg,

i've tested the PR.

* beagleboard untested (did not had a serial header)
* pandaboard broken u-boot (even before your patches, seems u-boot code is missing)
* beagleboard-xm broken linux (u-boot works, but no serial after jumping into kernel, will investigate)
* beagle bone black works

tldr: i'll merge your PR

best,
lynxis

**Comment by dhewg** [comment]
Huh, not too promising test results... did beaglebone-xm work before the changes?

**Comment by lynxis** [comment]
>  beaglebone-xm work before the changes?
No. I have to look with jtag into. Couldn't get early_printk work. It doesn't look like it's booting.

I've merged it. Thanks.
